File: market/test/java/ru/yandex/market/replenishment/autoorder/repository/RegionRepositoryTest.java
package ru.yandex.market.replenishment.autoorder.repository;

import java.util.List;

import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import ru.yandex.market.common.test.db.DbUnitDataSet;
import ru.yandex.market.replenishment.autoorder.config.FunctionalTest;
import ru.yandex.market.replenishment.autoorder.model.entity.postgres.Region;
import ru.yandex.market.replenishment.autoorder.repository.postgres.RegionRepository;

import static org.junit.jupiter.api.Assertions.assertEquals;
public class RegionRepositoryTest extends FunctionalTest {
    @Autowired
    RegionRepository regionRepository;

    @Test
    @DbUnitDataSet(before = "RegionRepositoryTest.testCheckRegionTypes.before.csv")
    public void testCheckRegionTypes() {
        List<Long> regionIds = List.of(1L, 2L, 3L);
        List<Long> regionTypes = List.of(4L, 5L);
        List<Region> regions = regionRepository.checkRegionTypes(regionIds, regionTypes);
        assertEquals(1, regions.size());
        assertEquals(2, regions.get(0).getId());
        assertEquals(0, regions.get(0).getParentId());
        assertEquals("region2", regions.get(0).getName());
    }
}

File: market/test/kotlin/ru/yandex/market/mbi/orderservice/tms/persistence/repository/RegionRepositoryTest.kt
package ru.yandex.market.mbi.orderservice.tms.persistence.repository

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.fail
import org.springframework.beans.factory.annotation.Autowired
import ru.yandex.market.common.test.db.DbUnitDataSet
import ru.yandex.market.mbi.orderservice.common.enum.RegionType
import ru.yandex.market.mbi.orderservice.tms.FunctionalTest
import ru.yandex.market.mbi.orderservice.common.persistence.repository.pg.RegionRepository

@DbUnitDataSet(before = ["region/regionRepositoryTest.before.csv"])
class RegionRepositoryTest : FunctionalTest() {

    @Autowired
    lateinit var regionRepository: RegionRepository

    @Test
    fun `test getting parent region`() {
        val roundToCountry = regionRepository.roundRegionTo(1, RegionType.COUNTRY) ?: fail("Region was expected")
        assertThat(roundToCountry).satisfies {
            assertThat(it.id).isEqualTo(3)
            assertThat(it.name).isEqualTo("Russia")
            assertThat(it.type).isEqualTo(RegionType.COUNTRY)
        }
    }
}
